src/vmspawn/vmspawn-register.c

```diff
@@ -5,36 +5,45 @@
 
 #include "bus-error.h"
 #include "bus-locator.h"
+#include "json.h"
 #include "macro.h"
 #include "process-util.h"
+#include "socket-util.h"
 #include "string-util.h"
+#include "varlink.h"
 #include "vmspawn-register.h"
 
-int register_machine(sd_bus *bus, const char *machine_name, sd_id128_t uuid, const char *service, const char *directory) {
-        _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
+int register_machine(
+                const char *machine_name,
+                sd_id128_t uuid,
+                const char *service,
+                const char *directory,
+                unsigned cid,
+                const char *address,
+                const char *key_path) {
+
+        _cleanup_(varlink_unrefp) Varlink *vl = NULL;
         int r;
 
-        assert(bus);
         assert(machine_name);
         assert(service);
 
-        r = bus_call_method(
-                        bus,
-                        bus_machine_mgr,
-                        "RegisterMachine",
-                        &error,
-                        NULL,
-                        "sayssus",
-                        machine_name,
-                        SD_BUS_MESSAGE_APPEND_ID128(uuid),
-                        service,
-                        "vm",
-                        (uint32_t) getpid_cached(),
-                        strempty(directory));
+        r = varlink_connect_address(&vl, "/run/systemd/machine/io.systemd.Machine");
```

Comment:
Please explain in the commit message why a D-bus call is replaced by a varlink call.